v2/tools/generator/internal/codegen/pipeline/create_arm_types.go

```diff
@@ -520,6 +521,28 @@ func (c *armTypeCreator) visitARMTypeName(
 		return astmodel.CreateARMTypeName(def.Name()), nil
 	}
 
+	// If the name is an alias for a primitive type, we look through it to the underlying type
```

Comment:
minor: Consider expanding this comment from the "what" we are doing, to the "why" (we flatten the types to break the circular dependency between ARM types and Kube types)